scripts/asm-installer/release_asm_installer

```diff
@@ -118,6 +127,28 @@ get_stable_version() {
   echo "${VER//+/-}"
 }
 
+get_version_file_and_lock() {
+  local FILE_PATH; FILE_PATH=${BUCKET_PATH}/${STABLE_VERSION_FILE}
+
+  if [[ ! $(curl -O "${BUCKET_URL}/${FILE_PATH}") ]]; then
+    echo "Create New Version File ${STABLE_VERSION_FILE}"
+    true >| ${STABLE_VERSION_FILE}
+    gsutil cp "${STABLE_VERSION_FILE}" gs://"${FILE_PATH}"
+  fi
+
+  gsutil retention ${HOLD_TYPE} set "gs://${FILE_PATH}"
+}
+
+upload_version_file_and_unlock() {
+  local FILE_PATH; FILE_PATH=${BUCKET_PATH}/${STABLE_VERSION_FILE}
```

Comment:
This is in multiple places. 1) you should quote "${BUCKET_PATH}/${STABLE_VERSION_FILE}" 2) consider putting it at the top of the file and making it `readonly`.

Reply:
Actually now I realize that this is already at the top of the file, so just `readonly` that and use it.

Reply:
Done. Forgot that I have defined that on the top..